File: cpp/Problem007.cpp
#include "EulerSolution.h"
#include "EulerUtils.h"
#include <string>
#include <cmath>

/*
 * @author Matthieu Le Boucher <matt.leboucher@gmail.com>
 */

using namespace std;

class Problem007 : public EulerSolution {
public:
    int number() override {
        return 7;
    }

    double solution() override {
        int currentNumber = 1;
        int primesCounter = 1;

        while (primesCounter < 10001) {
            currentNumber += 2;

            if (EulerUtils::is_prime(currentNumber))
                primesCounter++;
        }

        return currentNumber;
    }

    double answer() override {
        return 104743;
    }
};

File: java/Problem/Problem007.java
/*
 * This file contains a solution to the problem number 1 of the Project Euler, last modified on 5/9/2018.
 *
 * Copyright (c) Matthieu Le Boucher. All rights reserved.
 */

package Problem;

import Test.Solution;
import Utils.Arithmetics;

@Solution(number = 7, answer = 104743)
public class Problem007 implements EulerSolution {
    public double solution() {
        int currentNumber = 1;
        int primesCounter = 1;

        while (primesCounter < 10_001) {
            currentNumber += 2;

            if (Arithmetics.isPrime(currentNumber))
                primesCounter++;
        }

        return currentNumber;
    }

    public static void main(String[] args) {
        System.out.println((new Problem007()).solution());
    }
}
